### PR TITLE
New public blog setting 

### PR DESCRIPTION
**Pull request by dereksmart**
Do not merge until other pieces are in place. 

## Proposed Changes

Adds a new setting to the site's privacy page for wpcom and WoA sites, which opts out the site content from third parties and research partners. 

More info: peGwbA-1tL-p2

## Testing Instructions

On a wpcom simple and WoA site: 
- Head to /settings/general/
- Toggle `Prevent third party sharing` and save. Reloading the page should persist. 
- When the "Prevent third party sharing" option is checked, so should this one. It should not be possible to deselect the new setting when "Discourage search engines" is selected. 

You should not see this setting on a self-hosted Jetpack site. 

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by mdawaffe** [comment]
I know the wording is still being finalized. Logic makes sense to me.

**Review by taipeicoder — APPROVED** [review]
LGTM! 🚢

File: client/my-sites/site-settings/site-setting-privacy/form.tsx [hunk 1]
```diff
@@ -178,16 +194,14 @@ const SiteSettingPrivacyForm = ( {
 							<FormInputCheckbox
 								name="blog_public"
 								value="0"
-								checked={
-									( wpcomPublicComingSoon && blogPublic === 0 && isComingSoonDisabled ) ||
-									( 0 === blogPublic && ! wpcomPublicComingSoon )
-								}
+								checked={ discourageSearchChecked }
 								onChange={ () =>
 									handleVisibilityOptionChange( {
```
Comment (gravityrail) [inline client/my-sites/site-settings/site-setting-privacy/form.tsx:199]: Does this function actually handle the `wpcom_data_sharing_opt_out` value? It doesn't seem so.

**Review by gravityrail — APPROVED** [review]
LGTM

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/11488895

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Hi @dereksmart, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `Prevent third-party sharing for %(siteName)s`
- `This option will prevent this site’s content from being shared with our licensed network of content and research partners, including those that train AI models. {{a}}Learn more{{/a}}.`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/11488895).

**Pull request merged**